signer/src/main.rs

```diff
@@ -227,19 +227,36 @@ async fn run_shutdown_signal_watcher(ctx: impl Context) -> Result<(), Error> {
 async fn run_libp2p_swarm(ctx: impl Context) -> Result<(), Error> {
     tracing::info!("initializing the p2p network");
 
-    // Build the swarm.
     tracing::debug!("building the libp2p swarm");
     let config = ctx.config();
 
     let enable_quic = config.signer.p2p.is_quic_used();
 
+    // Limit the number of signers to the maximum number of signer pubkeys we
+    // can support. Note that this value is used as a base value for swarm
+    // connection limit calculations.
+    let num_signers = config
+        .signer
+        .bootstrap_signing_set()
+        .len()
+        .try_into()
+        .unwrap_or(signer::MAX_KEYS);
+
+    // This is how many seconds the swarm will wait before attempting to
+    // bootstrap (i.e. connect to other peers). Three seconds is a sane default
+    // value, giving the swarm a few seconds to start up and bind listener(s)
+    // before proceeding.
+    const INITIAL_BOOTSTRAP_DELAY_SECS: u64 = 3;
```

Comment:
nit/qq: should we move this const in a different place for visibility?

Reply:
sure, moved in [ae2e340](https://github.com/stacks-sbtc/sbtc/pull/1568/commits/ae2e340a7f6687ad9992599902df47ef8c96de79)